### PR TITLE
MINOR: Remove deprecated valueTransformer.punctuate

### PR DESCRIPTION
**Pull request by guozhangwang**
Also removed the InternalValueTransformerWithKey / Supplier which is used to mock away the deprecated punctuate function.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@mjsax this is ready for reviews now.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 1]
```diff
@@ -49,8 +49,8 @@
     public void testToInternlValueTransformerSupplierSuppliesNewTransformers() {
         final ValueTransformerSupplier vts = createMock(ValueTransformerSupplier.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: Nit: rename `vts`

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 2]
```diff
@@ -49,8 +49,8 @@
...
         expect(vts.get()).andReturn(null).times(3);
-        final InternalValueTransformerWithKeySupplier ivtwks =
-            AbstractStream.toInternalValueTransformerSupplier(vts);
+        final ValueTransformerWithKeySupplier ivtwks =
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: nit: rename `ivtwks`

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java [hunk 3]
```diff
@@ -60,15 +60,12 @@ public void testToInternlValueTransformerSupplierSuppliesNewTransformers() {
 
     @Test
     public void testToInternalValueTransformerSupplierSuppliesNewTransformers() {
-        final ValueTransformerWithKeySupplier vtwks =
-            createMock(ValueTransformerWithKeySupplier.class);
+        final ValueTransformerWithKeySupplier vtwks = createMock(ValueTransformerWithKeySupplier.class);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/AbstractStreamTest.java]: as above

**Review by mjsax — APPROVED** [review]
Thanks for the PR. Overall LGTM. Some nits. Feel free to merge.

**Pull request merged**